pages/configuration/elasticsearch.rst

```diff
@@ -8,108 +8,77 @@ We strongly recommend to use a dedicated Elasticsearch cluster for your Graylog
-list of Elasticsearch nodes to ``elasticsearch.yml``::
-
-  discovery.zen.ping.multicast.enabled: false
-  discovery.zen.ping.unicast.hosts: ["es-node-1.example.org:9300" , "es-node-2.example.org:9300"]
-
-The Elasticsearch default communication port is *9300/tcp* (not to be confused with the HTTP interface running on port *9200/tcp* by default).
-
-The communication port can be changed in the Elasticsearch configuration file (``elasticsearch.yml``) with the configuration setting ``transport.tcp.port``.
-
-Last but not least, make sure that Elasticsearch is binding to a network interface that Graylog can connect to (see ``network.host`` and `Commonly Used Network Settings <https://www.elastic.co/guide/en/elasticsearch/reference/2.3/modules-network.html#common-network-settings>`_).
-
+Graylog uses automatic node discovery to gather a list of all available Elasticsearch nodes in the cluster at runtime and distribute requests among them to potentially increase performance and availability. To enable this feature, you need to set the ``elasticsearch_discovery_enabled`` to ``true``. Optionally, you can define the a filter allowing to selectively include/exclude discovered nodes (details how to specify node filters are found in the `Elasticsearch documentation <https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster.html#cluster-nodes>`_) using the ``elasticsearch_discovery_filter`` setting, or tuning the frequency of the node discovery using the ``elasticsearch_discovery_frequency`` configuration option.
 
 Configuration of Elasticsearch nodes
 ------------------------------------
 
-Disable dynamic scripting
-^^^^^^^^^^^^^^^^^^^^^^^^^
-
-Elasticsearch prior to version 1.2 had an `insecure default configuration <http://bouk.co/blog/elasticsearch-rce/>`__ which could lead to a `remote code execution <https://groups.google.com/forum/#!msg/graylog2/-icrS0rIA-Q/cCTJaNjVrQAJ>`__.
-
-Make sure to add the following settings to the ``elasticsearch.yml`` file to disable the dynamic scripting feature and
-prevent possible remote code executions::
-
-  script.inline: false
-  script.indexed: false
-  script.file: false
-
-Details about dynamic scripting can be found in `the reference documentation of Elasticsearch <https://www.elastic.co/guide/en/elasticsearch/reference/2.3/modules-scripting.html>`__.
-
 Control access to Elasticsearch ports
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Since Elasticsearch has no authentication mechanism at time of this writing, make sure to restrict access to the Elasticsearch
-ports (default: 9200/tcp and 9300/tcp). Otherwise the data is readable by anyone who has access to the machine over network.
+If you are not using `Shield/X-Pack <https://www.elastic.co/products/x-pack/security>`_ to authenticate access to your Elasticsearch nodes, make sure to restrict access to the Elasticsearch ports (default: 9200/tcp and 9300/tcp). Otherwise the data is readable by anyone who has access to the machine over network.
```

Comment:
✅ Added Search Guard. I think it is valid to promote tools like that. We are not competing with them and they add value to our product as well.